Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,505 @@
+
+            if tdef and len(self.torrents) < self.max_torrents:
+                # Create a torrent dict.
+                real_infohash = tdef.get_infohash()
+                torrent_values = [item_torrent['title'], tdef, tdef.get_creation_date(), tdef.get_length(),
+                                  len(tdef.get_files()), -1, -1, self.enabled, {}]
+
+                # store the real infohash to generated infohash
+                self.torrents[real_infohash] = dict(zip(torrent_keys, torrent_values))
+                self.fake_infohash_id[sha1(item_torrent['url']).digest()] = real_infohash
+
+                # manually generate an ID and put this into DB
+                self.torrent_db.addOrGetTorrentID(real_infohash)
+                self.torrent_db.addExternalTorrent(tdef)
+
+                # create Torrent object and store it
+                self.torrent_mgr.load_torrent(Torrent.fromTorrentDef(tdef))
+
+                # Notify the BoostingManager and provide the real infohash.
+                if self.torrent_insert_callback:
+                    self.torrent_insert_callback(self.source, real_infohash, self.torrents[real_infohash])
+            elif tdef:
+                self._logger.debug("Max torrents in source reached. Not adding %s", tdef.get_infohash())
+
+        def __success_cb(response, item_dict):
+            return readBody(response).addCallback(__cb_body, item_dict).addErrback(self._on_err)
+
+        regex_unescape_xml = re.compile(r"\&\#(x?[0-9a-fA-F]+);")
+
+        for item in feed_status['items']:
+            # Not all RSS feeds provide us with the infohash,
+            # so we use a fake infohash based on the URL to identify the torrents.
+            url = regex_unescape_xml.sub(ent2chr, item['url'])
+            fake_infohash = sha1(url).digest()
+            if fake_infohash not in self.fake_infohash_id.keys():
+                # create Agent to download torrent file
+                self.fake_infohash_id[fake_infohash] = None
+                agent = Agent(reactor)
+                ses_agent = agent.request(
+                    'GET',  # http://stackoverflow.com/a/845595
+                    urllib.quote(url, safe="%/:=&?~#+!$,;'@()*[]"),
+                    Headers({'User-Agent': ['Tribler ' + version_id]}),
+                    None)
+                ses_agent.addCallback(__success_cb, item).addErrback(self._on_err)
+
+    def kill_tasks(self):
+        BoostingSource.kill_tasks(self)
+
+
+class DirectorySource(BoostingSource):
+    """
+    Credit mining source from a local directory
+
+    The directory must exist.
+    """
+
+    def __init__(self, session, directory, boost_settings, torrent_insert_cb):
+        BoostingSource.__init__(self, session, directory, boost_settings, torrent_insert_cb)
+
+    def _load(self, directory):
+        if os.path.isdir(directory):
+            # Wait for __init__ to finish so the source is registered with the
```

Comment:
Yes, it's been like that since your branch. When I want to remove it, somehow it will pop an error. I was looking for a fix but still didn't work.

https://github.com/whirm/tribler/blob/channel_boosting_new_exp_rebase/Tribler/Policies/BoostingManager.py#L663